src/python/WMCore/BossAir/Plugins/PyCondorPlugin.py

```diff
@@ -954,44 +925,44 @@ def customizePerJob(self, job):
         if job.get('proxyPath'):
             jdl.append('x509userproxy = %s\n' % job['proxyPath'])
 
-        if job.get('requestName'):
-            jdl.append('+WMAgent_RequestName = "%s"\n' % job['requestName'])
-            m = GROUP_NAME_RE.match(job['requestName'])
-            if m:
-                jdl.append('+CMSGroups = %s\n' % classad.quote(m.groups()[0]))
-
-        if job.get('taskName'):
-            jdl.append('+WMAgent_SubTaskName = "%s"\n' % job['taskName'])
+        jdl.append('+WMAgent_RequestName = "%s"\n' % job['requestName'])
+        match = GROUP_NAME_RE.match(job['requestName'])
+        if match:
+            jdl.append('+CMSGroups = %s\n' % classad.quote(match.groups()[0]))
+        else:
+            jdl.append('+CMSGroups = undefined\n')
 
-        if job.get('taskType'):
-            jdl.append('+CMS_JobType = "%s"\n' % job['taskType'])
+        jdl.append('+WMAgent_SubTaskName = "%s"\n' % job['taskName'])
+        jdl.append('+CMS_JobType = "%s"\n' % job['taskType'])
 
         # Handling for AWS, cloud and opportunistic resources
         jdl.append('+AllowOpportunistic = %s\n' % job.get('allowOpportunistic', False))
 
         # dataset info
         if job.get('inputDataset'):
             jdl.append('+DESIRED_CMSDataset = "%s"\n' % job['inputDataset'])
+        else:
+            jdl.append('+DESIRED_CMSDataset = undefined\n')
         if job.get('inputDatasetLocations'):
             jdl.append('+DESIRED_CMSDataLocations = "%s"\n' % ','.join(job['inputDatasetLocations']))
+        else:
+            jdl.append('+DESIRED_CMSDataLocations = undefined\n')
 
         # HighIO jobs
-        highIOjob = job.get('highIOjob')
-        if highIOjob:
-            jdl.append('+Requestioslots = 1\n')
-
-        # Performance estimates
-        if job.get('estimatedJobTime'):
-            jdl.append('+MaxWallTimeMins = %d\n' % int(job['estimatedJobTime'] / 60.0))
-        if job.get('estimatedMemoryUsage'):
-            jdl.append('request_memory = %d\n' % int(job['estimatedMemoryUsage']))
-        if job.get('estimatedDiskUsage'):
-            jdl.append('request_disk = %d\n' % int(job['estimatedDiskUsage']))
-
-        # Set up JDL for multithreaded jobs
-        if job.get('numberOfCores', 1) > 1:
-            jdl.append('machine_count = 1\n')
-            jdl.append('request_cpus = %s\n' % job.get('numberOfCores', 1))
+        jdl.append('+Requestioslots = %d\n' % job.get('highIOjob', 0))
+
+        # Performance and resource estimates
+        numberOfCores = job.get('numberOfCores', 1)
+        requestMemory = int(job['estimatedMemoryUsage']) if job.get('estimatedMemoryUsage', None) else 1000
+        requestDisk = int(job['estimatedDiskUsage']) if job.get('estimatedDiskUsage', None) else 20*1000*1000*numberOfCores
+        maxWallTimeMins = int(job['estimatedJobTime'])/60.0 if job.get('estimatedJobTime', None) else 12*60
```

Comment:
Hmm, I had this impression :) I agree with your first argument and playing safe is better. However, it will cause pilot inefficiencies (25% cause by walltime) and may be even worse for HLT farm and future opportunistic resources.

I'm just wondering whether it would be worthy to have 1 problematic job out of 1M, but taking the most efficiency approach. OR, having 0 problems but injecting not negligible efficiency to gwms... (?)